docker/build-image.sh

```diff
@@ -44,9 +44,61 @@ while [[ $# -gt 0 ]]; do
             COMPONENT_VERSION_STRING=develop
             shift
             ;;
+        --ros-1-packages)
+            ROS1_PACKAGES=""
+            shift
+            ;;
+        --ros-2-packages)
+            ROS2_PACKAGES=""
+            shift
+            ;;
+        *)
```

Comment:
I don't suppose there's anyway to notify the user if these options were provided out of order?

Reply:
I'll look into it, the problem mostly comes in with how I'm accumulating the lists below, once it sees --ros-2-packages it assumes means that --ros-1-packages has already been processed and that it can consume all further inputs. It might be hard to differentiate between the case where --ros-1-packages is not set, and the case where it is set second. Maybe I can check if one of the options has been specified, but no packages were provided and throw an error there?